extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/impl/DbCheckingPMetaSupplier.java

```diff
@@ -0,0 +1,195 @@
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.security.Permission;
+import java.util.List;
+import java.util.Map.Entry;
+import java.util.function.Function;
+
+import static com.hazelcast.internal.util.UuidUtil.newUnsecureUuidString;
+import static com.hazelcast.jet.datamodel.Tuple2.tuple2;
+import static com.hazelcast.jet.impl.util.Util.arrayIndexOf;
+import static com.hazelcast.jet.mongodb.impl.MongoUtilities.checkCollectionExists;
+import static com.hazelcast.jet.mongodb.impl.MongoUtilities.checkDatabaseExists;
+import static com.hazelcast.partition.strategy.StringPartitioningStrategy.getPartitionKey;
+import static java.util.Collections.singletonList;
+
+/**
+ * A {@link ProcessorMetaSupplier} that will check if requested database and collection exist before creating
+ * the processors.
+ */
+public class DbCheckingPMetaSupplier implements ProcessorMetaSupplier {
+
+    private final Permission requiredPermission;
+    private final boolean shouldCheck;
+    private boolean forceTotalParallelismOne;
+    private final String databaseName;
+    private final String collectionName;
+    private final ProcessorSupplier processorSupplier;
+    private final SupplierEx<? extends MongoClient> clientSupplier;
+    private final DataConnectionRef dataConnectionRef;
+    private int preferredLocalParallelism = Vertex.LOCAL_PARALLELISM_USE_DEFAULT;
+
+    private transient Address ownerAddress;
+
+    /**
+     * Creates a new instance of this meta supplier.
+     */
+    public DbCheckingPMetaSupplier(@Nullable Permission requiredPermission,
+                                   boolean shouldCheck,
+                                   boolean forceTotalParallelismOne,
+                                   @Nullable String databaseName,
+                                   @Nullable String collectionName,
+                                   @Nullable SupplierEx<? extends MongoClient> clientSupplier,
+                                   @Nullable DataConnectionRef dataConnectionRef,
+                                   @Nonnull ProcessorSupplier processorSupplier) {
+        this.requiredPermission = requiredPermission;
+        this.shouldCheck = shouldCheck;
+        this.forceTotalParallelismOne = forceTotalParallelismOne;
+        this.databaseName = databaseName;
+        this.collectionName = collectionName;
+        this.processorSupplier = processorSupplier;
+        this.clientSupplier = clientSupplier;
+        this.dataConnectionRef = dataConnectionRef;
+    }
+
+    /**
+     * Sets preferred local parallelism. If {@link #forceTotalParallelismOne} is selected, this
+     * method will have no effect.
+     */
+    public DbCheckingPMetaSupplier withPreferredLocalParallelism(int preferredLocalParallelism) {
+        this.preferredLocalParallelism = forceTotalParallelismOne ? 1 : preferredLocalParallelism;
+        return this;
+    }
+
+    @Override
+    public int preferredLocalParallelism() {
```

Comment:
why not use composition with regular `forceTotalParallelismOne` PMS?

Reply:
Good idea, cf7443721f797f74b4e4018436a055219c98c10f

Reply:
can you just wrap DbCheckingPMetaSupplier with forceTotalParallelismOne PMS? You would need to add variant of PMS.forceTotalParallelismOne method that takes another PMS

Reply:
Will be there any benefit? Right now it's changes so I use composition.